### PR TITLE
Unit test for container fields and configmap fields

### PR DESCRIPTION
**Pull request by ZichengMa**
Finish all the tests for set functions in container, configmap, and some other types as needed.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (marshtompsxd) [inline .github/workflows/ci.yml:1]: What are these changes?

**Comment by marshtompsxd** [comment]
@ZichengMa You might want to solve the conflicts first. Obviously you didn't start from the current main branch.

**Comment by marshtompsxd** [comment]
@ZichengMa You can probably just close this PR and create another one and copy your tests there.

**Comment by ZichengMa** [comment]
> @ZichengMa You can probably just close this PR and create another one and copy your tests there.

OK. I will fix it and more tests soon.

**Pull request closed without merging**